mlx/fast.cpp

```diff
@@ -392,8 +392,8 @@ array rope(
           s);
     };
 
-    auto inv_freqs =
-        inputs.size() == 3 ? reciprocal(inputs[2], s) : default_inv_freqs();
+    auto inv_freqs = inputs.size() == 3 ? astype(reciprocal(inputs[2], s), t)
```

Comment:
So presumably this is for precision that we don't do `reciprocal(astype(inputs[2], t, s), s)` ?

Also small issue here is that `astype` should also get the stream.

```suggestion
    auto inv_freqs = inputs.size() == 3 ? astype(reciprocal(inputs[2], s), t, s)
```

Reply:
Yes it seemed safer but it's not really based on anything concrete. Good catch with the stream!